client/ayon_maya/plugins/create/create_animation_pointcache.py

```diff
@@ -132,3 +140,203 @@ def create(self, product_name, instance_data, pre_create_data):
+            "creator_attributes": {},
+            "publish_attributes": {},
+        }
+
+        # Collect transient data of where the asset is in the USD Stage
+        # hierarchy and the reference node belonging to it.
+        created_instance = CreatedInstance.from_existing(node_data, self)
+        created_instance.transient_data["reference_node"] = reference_node
+        self._add_instance_to_context(created_instance)
+        return created_instance
+
+    def _collect_transient_data(self, created_instance):
+        import ufe
+
+        # Find the reference node
+        reference_node = created_instance.transient_data.get("reference_node")
+        if not reference_node:
+            # Existing scene instance
+            reference_node = created_instance["instance_node"]
+
+        dag_path = self.get_pulled_maya_reference_dag_path(reference_node)
+        pulled_info = self.get_pulled_info(dag_path)
+        path_string = ufe.PathString.string(pulled_info.pulled_path)
+        dag_shape, prim_path = path_string.split(",", 1)
+
+        # TODO: This should instead find the nearest asset instead of assuming
+        #   by splitting strings
+        # split off /rig/rigMain
+        target_prim_path = prim_path.rsplit("/", 2)[0]
+        created_instance.transient_data["target_prim_path"] = target_prim_path
+
+    def update_instances(self, update_list):
+        # We only generate the persisting layer data into the scene once
+        # we save with the UI on e.g. validate or publish
+        for instance, _changes in update_list:
+            instance_node = instance.data.get("instance_node")
+
+            # Ensure a node exists to persist the data to
+            if not instance_node:
+                instance_node = self._create_instance_node(instance)
+                instance.data["instance_node"] = instance_node
+
+            self.imprint_instance_node(instance_node,
+                                       data=instance.data_to_store())
+
+    def _create_instance_node(self, instance):
+        """Create object set from product name"""
+        reference_node = instance.transient_data.get("reference_node")
+        members = [reference_node] if reference_node else []
+        return cmds.sets(members, name=instance.data["productName"])
+
+    @staticmethod
+    def get_pulled_maya_reference_dag_path(reference_node) -> Optional[str]:
+        """Return the maya dag path to the root node of a MayaReference prim
+        in a Maya USD Proxy Shape that is currently in edit mode (is currently
+        a loaded reference)"""
+        try:
+            import mayaUsdUtils
+        except ImportError:
+            # No maya usd plug-in
+            return
+
+        associated_nodes = cmds.listConnections(
+            f"{reference_node}.associatedNode")
+        if not associated_nodes:
+            return
+
+        for node in associated_nodes:
+            if mayaUsdUtils.isPulledMayaReference(node):
```

Comment:
```suggestion
            if mayaUsdUtils.isPulledMayaReference(node):
```
I strangely hit this error when I tested with maya 2025.1.
![image](https://github.com/user-attachments/assets/a5addc46-3b5b-4725-a2f7-caf87ea916cf)

Reply:
Thanks. I did notice some issues in Maya 2025 with regards to their USD API too.

I also saw this in their recent [release notes of Maya 2025.3](https://help.autodesk.com/view/MAYAUL/2025/ENU/?guid=GUID-59686914-791F-4386-8B0C-78DBEA96B029):

![image](https://github.com/user-attachments/assets/6d6225a3-74c8-4e44-a970-1ebe064126b1)

Which is also something I'll need to double check.